Neo4j.Driver/Neo4j.Driver/V1/TrustManager.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) 2002-2018 Neo4j Sweden AB [http://neo4j.com]
+// 
+// This file is part of Neo4j.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Neo4j.Driver.Internal.Connector.Trust;
+
+namespace Neo4j.Driver.V1
+{
+
+    /// <summary>
+    /// This is the base class all built-in or custom trust manager implementations should be inheriting from. Trust managers
+    /// are the way that one could customise how TLS trust is established.
+    /// </summary>
+    public abstract class TrustManager
+    {
+
+        internal ILogger Logger { get; set; }
+
+        /// <summary>
+        /// Returns whether the endpoint should be trusted or not.
+        /// </summary>
+        /// <param name="uri">The uri towards which we're establishing connection</param>
+        /// <param name="certificate">The certificate presented by the other endpoint</param>
+        /// <param name="chain">The certificate chain that was built during the handshake</param>
+        /// <param name="sslPolicyErrors">The initial policy errors that shows what problems were detected during the handshake</param>
+        /// <returns><value>true</value> if the connection should be established, <value>false</value> otherwise</returns>
+        public abstract bool ValidateServerCertificate(Uri uri, X509Certificate2 certificate, X509Chain chain,
+            SslPolicyErrors sslPolicyErrors);
+
+        /// <summary>
+        /// Creates an insecure trust manager that trusts any certificate it is presented, but does hostname verification.
+        /// </summary>
+        /// <returns>An instance of <see cref="TrustManager"/></returns>
+        public static TrustManager CreateInsecure() => CreateInsecure(false);
```

Comment:
Seems different from docs. The hostname verification is turned off by default for this one.
Not sure if we shall turn it on by default.
Maybe worth to test on linux and macOs machines too?

Reply:
Is this feature fully available on all platforms?